### ISSUE TITLE
[MU4 Issue] Application seems to lose/gain focus (highlight) while traversing the top text menu items

### ISSUE THREAD

**Issue by worldwideweary** [musescore/MuseScore#10681]
**Describe the bug**
At least on my Linux distro (Mint - Release: Una), the application title and the "V" icon (for max/minimize etc)  flashes while mousing through the file-type menu, (and the application icon in taskbar selects/deselects concurrently):

https://user-images.githubusercontent.com/7139517/155895463-12c22a07-0473-4836-b082-bb44a012a62b.mp4

It would seem that MuseScore is losing and regaining focus in the process.

**Comment by cbjeukendrup** [comment]
Indeed, whenever a menu, popup or tooltip is opened, the main window looses focus. This should be solvable since for example Qt Creator's menus (which are also non-native) don't show this behaviour. (of course, 'under the hood', the popup window _should_ get focus, but it should not have a visual effect.)

**Comment by llui85** [comment]
Possibly related to #10218?

**Comment by tibequadorian** [comment]
I noticed that in a ~15 sec session where I heavily mouse through the panel, some functions in the debug output are more often called and have significantly more sum time:
```
Function                                                    Call time         Call count        Sum time
NavigationController::onActiveRequested                     0.169 ms          110               18.547 ms
UiActionsRegister::updateEnabledAll                         0.121 ms          112               13.575 ms
NavigationControl::setPanel                                 0.002 ms          1223              2.601 ms
NavigationController::doDeactivateSection                   0.020 ms          75                1.482 ms
NavigationController::doDeactivatePanel                     0.015 ms          80                1.175 ms
NavigationController::doDeactivateControl                   0.001 ms          747               0.873 ms
UiContextResolver::currentUiContext                         0.006 ms          113               0.709 ms
```
compared to an idle session of the same length:
```
Function                                                    Call time         Call count        Sum time
NavigationController::onActiveRequested                     0.143 ms          1                 0.143 ms
UiActionsRegister::updateEnabledAll                         0.086 ms          3                 0.259 ms
UiContextResolver::currentUiContext                         0.003 ms          4                 0.012 ms
```
I hope this helps.

**Comment by Tantacrul** [comment]
> Indeed, whenever a menu, popup or tooltip is opened, the main window looses focus. This should be solvable since for example Qt Creator's menus (which are also non-native) don't show this behaviour. (of course, 'under the hood', the popup window _should_ get focus, but it should not have a visual effect.)

Is this is still an issue?

**Comment by Tantacrul** [comment]
@worldwideweary - same question :)

**Comment by tibequadorian** [comment]
Yes it is, tested with [MuseScoreNightly-202205310429-master-7eb232e-x86_64.AppImage](https://ftp.osuosl.org/pub/musescore-nightlies/linux/4x/nightly/MuseScoreNightly-202205310429-master-7eb232e-x86_64.AppImage)

**Comment by tibequadorian** [comment]
https://user-images.githubusercontent.com/9560587/171219715-c34f6f91-fa09-4b74-b13a-365e0b79408c.mp4

**Comment by tibequadorian** [comment]
Great, your PR solves this issue for me @Eism

**Comment by worldwideweary** [comment]
@Tantacrul Hey this appears to be resolved on my Linux distro

**Comment by Tantacrul** [comment]
OK, @DmitryArefiev: this issue looks like it can be closed. Feel free to reopen if you can still reproduce it.

**Comment by cbjeukendrup** [comment]
On macOS, it is solved with menus and tooltips, but when opening a popup, the main window still loses focus. I'm not sure if that's desired behaviour, but at least I don't like it.

**Comment by Eism** [comment]
For popups, focus is needed, since there may be components that need focus (for example, a text input field).
Are there any examples of programs where such pop-ups open without changing the focus?

**Comment by Eism** [comment]
Another question is how critical it is for macOS?
I see that only the buttons from the title bar become inactive.

**Comment by Tantacrul** [comment]
I have the same question: what is the harm?

**Comment by cbjeukendrup** [comment]
For example, in Qt Creator this does not occur with the project / build type selector popup. That thing is also implemented as a separate window, similar to our popups:

https://user-images.githubusercontent.com/48658420/171594643-b9324ea6-2bdd-4fde-a0ea-af498fdd0f0c.mov

Also, with native macOS popups this also never occurs, and those are also implemented as separate windows. 

It is relatively harmless, but just awful and slightly annoying.

**Comment by Eism** [comment]
In your Qt Creator example, there are no components that need focus. 
It's more like our menus(without focus).

**Comment by cbjeukendrup** [comment]
Ok, let's close it then for now. If I ever incidentally find out how macOS popups do it, I can try to apply that to our popups too. But it shouldn't be a priority.

**Comment by tibequadorian** [comment]
For me the lagging menu selection is still an issue, see: https://github.com/musescore/MuseScore/issues/10681#issuecomment-1142331995
and it got only fixed by #11750 so you should look into merging that.

**Comment by boryskuzmenko** [comment]
🟢 **FIXED**
Tested on macOS 12, Win 11, Ubuntu 20
